Makefile

```diff
@@ -284,7 +284,7 @@ ifdef LUA
     # Windows expects to have lua unpacked at a specific location
     LDFLAGS += -llua
   else
-    LUA_CANDIDATES = lua5.2 lua-5.2 lua5.1 lua-5.1 lua
+    LUA_CANDIDATES = $(basename $(LUA_BINARY)) lua5.2 lua-5.2 lua5.1 lua-5.1 lua
```

Comment:
That's not what I asked, did you encounter this problem, or are you
predicting that it will happen?
I ask because this change *will* break some builds, since by default this
will build against lua preferentially, when it needs to build against
lua5.2 when available.